### ISSUE TITLE
Difficulty adjustment failed on shard chains in testnet 2

### ISSUE THREAD

**Issue by qcgg** [QuarkChain/pyquarkchain#237]
The difficulty is stuck below 512 and blocks are mined too fast.
The root cause is that the minimum difficulty isn't explicitly set and the default value 1 is used.
https://github.com/QuarkChain/pyquarkchain/blob/699437912bcb662f7356b30ecaf67a630281bc9b/quarkchain/cluster/shard_state.py#L67
https://github.com/QuarkChain/pyquarkchain/blob/699437912bcb662f7356b30ecaf67a630281bc9b/quarkchain/diff.py#L7

Fixing the code will cause hard fork. We'd rather fix the code and start a new testnet.

**Comment by qcgg** [comment]
Mining results for current testnet (network_id=21)
```
$ python ../testnet/block_owners.py --db=data/master.db --root_height=960
Counting shard blocks from root block height 960

ETHASH has 31330 blocks
314b2cd22c6d26618ce051a58c65af1253aecbb8: 43.64%
248dc97675f46cb2aeca53006f647ed94ef5b502: 25.23%
0000000000000000000000000000000000000000: 17.75%
b16a444c9a0d68f5e9f60e1004091b9f0cda34a9: 10.88%
02772c614f2c339e87527e75e63859bb6adeaa36: 1.12%
4363b57f2eff8acac6a2b4896e4ad3905b1029d9: 0.42%
557fd639ed70ad9dcbff0007958e9004cdb71e80: 0.36%
db7fd07891697f74a7e5102cc2cc522c25dc06e9: 0.23%
4bfc9294f73c03a680eff9d9432486affc1f815d: 0.12%
63364490cbffa90df288bb396cc7f6e769ca8fab: 0.10%
8d5153032ae3fe11d63b544e171aeff38988b874: 0.07%
8a82db4770318fdfd591b803ef06141415690663: 0.06%
626d9432723f2a6a92b0d0a2e2d917f5711ea6cd: 0.03%

SHA2SHA2 has 17077 blocks
314b2cd22c6d26618ce051a58c65af1253aecbb8: 40.37%
0000000000000000000000000000000000000000: 26.74%
248dc97675f46cb2aeca53006f647ed94ef5b502: 17.65%
b16a444c9a0d68f5e9f60e1004091b9f0cda34a9: 11.66%
557fd639ed70ad9dcbff0007958e9004cdb71e80: 1.49%
63364490cbffa90df288bb396cc7f6e769ca8fab: 1.24%
f0d1d032719b783bf3ff03eabe12cfcc6ad6e770: 0.44%
db7fd07891697f74a7e5102cc2cc522c25dc06e9: 0.21%
626d9432723f2a6a92b0d0a2e2d917f5711ea6cd: 0.15%
02772c614f2c339e87527e75e63859bb6adeaa36: 0.05%

QKCHASH has 15604 blocks
314b2cd22c6d26618ce051a58c65af1253aecbb8: 43.26%
0000000000000000000000000000000000000000: 20.42%
6bd55c9c9d54dc7973a760885eaef778ac821898: 16.75%
248dc97675f46cb2aeca53006f647ed94ef5b502: 9.78%
52bec11bf55be5235cabb148fba7a0ea0efad2d9: 4.64%
f0d1d032719b783bf3ff03eabe12cfcc6ad6e770: 2.10%
b16a444c9a0d68f5e9f60e1004091b9f0cda34a9: 2.06%
63364490cbffa90df288bb396cc7f6e769ca8fab: 0.96%
db7fd07891697f74a7e5102cc2cc522c25dc06e9: 0.03%
```

**Comment by qcdll** [comment]
fixed by #238
